File: Code/apps/problems_directory/problems/fizz-buzz/drivers/javascript/driver.js
const fs = require('fs');
const DELIMITER = '$$$DELIMITER$$$';

try {
    const inputData = fs.readFileSync('/app/input.txt', 'utf-8');
    // Split by delimiter and filter empty entries
    const testCases = inputData.split(DELIMITER).map(tc => tc.trim()).filter(tc => tc !== '');

    testCases.forEach((testCase) => {
        const lines = testCase.split('\n').map(l => l.trim()).filter(l => l !== '');
        
        const arg0 = Number(lines[0]);

        const result = fizzBuzz(arg0);
        
        // Output JSON stringified result
        console.log(JSON.stringify(result));
        console.log(DELIMITER);
    });
} catch (err) {
    console.error(err);
}

File: Code/apps/problems_directory/problems/fizz-buzz/drivers/cpp/driver.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <algorithm>
#include <fstream> 

using namespace std;

// USER CODE WILL BE INJECTED HERE
vector<string> fizzBuzz(int);

template <typename T>
void printResult(const vector<T>& v) {
    cout << "[";
    for (size_t i = 0; i < v.size(); ++i) {
        cout << v[i];
        if (i < v.size() - 1) cout << ",";
    }
    cout << "]";
}

template <typename T>
void printResult(const T& val) {
    if constexpr (is_same_v<T, bool>) {
        cout << (val ? "true" : "false");
    } else if constexpr (is_same_v<T, string>) {
        cout << "\"" << val << "\"";
    } else {
        cout << val;
    }
}

int main() {
    const string DELIMITER = "$$$DELIMITER$$$";
    
    stringstream buffer;
    buffer << cin.rdbuf(); 
    string content = buffer.str();

    if (content.empty()) {
        ifstream t("/app/input.txt");
        if(t.is_open()) {
            stringstream fbuffer;
            fbuffer << t.rdbuf();
            content = fbuffer.str();
        }
    }

    size_t prev = 0;
    while (true) {
        size_t pos = content.find(DELIMITER, prev);
        string testCase = (pos != string::npos) ? content.substr(prev, pos - prev) : content.substr(prev);
        
        testCase.erase(0, testCase.find_first_not_of(" \n\r\t"));
        testCase.erase(testCase.find_last_not_of(" \n\r\t") + 1);

        if (!testCase.empty()) {
            stringstream ss(testCase);
            int arg0; ss >> arg0;
            
            auto result = fizzBuzz(arg0);
            
          

            printResult(result);
            cout << endl << DELIMITER << endl;
        }

        if (pos == string::npos) break;
        prev = pos + DELIMITER.length();
    }
    return 0;
}